build.gradle.kts

```diff
@@ -225,12 +227,37 @@ tasks {
 	}
 
 	javadoc {
-		// of all the javadoc checks (accessibility, html, missing, reference, syntax; see
-		// https://docs.oracle.com/javase/8/docs/technotes/tools/unix/javadoc.html#BEJEFABE)
-		// disable the warning for missing comments and tags because they spam the output
-		// (it does often not make sense to comment every tag; e.g. the @return tag on annotations)
-		(options as CoreJavadocOptions).addStringOption("Xdoclint:accessibility,html,syntax,reference", "-quiet")
-		options.encoding = "UTF-8"
+		javadocTool.set(project.javaToolchains.javadocToolFor {
+			// Create Javadoc with newer JDK to get the latest features, e.g. search bar
+			languageVersion.set(JavaLanguageVersion.of(17))
+		})
+
+		// Exclude internal implementation package from javadoc
+		/*
+		 * TODO: Currently disabled because modular Gradle build fails when this package is excluded
+		 * See https://stackoverflow.com/q/32785002 and slightly related Gradle issue https://github.com/gradle/gradle/issues/14066
+		*/
+		//exclude("org/junitpioneer/internal")
```

Comment:
Would this help?

```suggestion
		if (!modularBuild.toBoolean()) {
			exclude("org/junitpioneer/internal")
		}
```

Reply:
Yes that works, and is probably a better solution for now, have changed this.